acvm/src/pwg/blackbox/schnorr_verify.rs

```diff
@@ -0,0 +1,81 @@
+use acir::{
+    circuit::opcodes::FunctionInput,
+    native_types::{Witness, WitnessMap},
+    BlackBoxFunc, FieldElement,
+};
+
+use crate::{
+    pwg::{insert_value, witness_to_value, OpcodeResolution, OpcodeResolutionError},
+    BlackBoxFunctionSolver,
+};
+
+pub(super) fn schnorr_verify(
+    backend: &impl BlackBoxFunctionSolver,
+    initial_witness: &mut WitnessMap,
+    public_key_x: &FunctionInput,
+    public_key_y: &FunctionInput,
+    signature: &[FunctionInput],
+    message: &[FunctionInput],
+    output: &Witness,
+) -> Result<OpcodeResolution, OpcodeResolutionError> {
+    // In barretenberg, if the signature fails, then the whole thing fails.
+
+    let public_key_x: &FieldElement = witness_to_value(initial_witness, public_key_x.witness)?;
+    let public_key_y: &FieldElement = witness_to_value(initial_witness, public_key_y.witness)?;
+
+    let mut signature = signature.iter();
+    let mut sig_s = [0u8; 32];
+    for (i, sig) in sig_s.iter_mut().enumerate() {
+        let _sig_i = signature.next().ok_or_else(|| {
+            OpcodeResolutionError::BlackBoxFunctionFailed(
+                BlackBoxFunc::SchnorrVerify,
+                format!("sig_s should be 32 bytes long, found only {i} bytes"),
+            )
+        })?;
+        let sig_i = witness_to_value(initial_witness, _sig_i.witness)?;
+        *sig = *sig_i.to_be_bytes().last().ok_or_else(|| {
+            OpcodeResolutionError::BlackBoxFunctionFailed(
+                BlackBoxFunc::SchnorrVerify,
+                "could not get last bytes".into(),
+            )
+        })?;
+    }
+    let sig_s: FieldElement = FieldElement::from_be_bytes_reduce(&sig_s);
+
+    let mut sig_e = [0u8; 32];
+    for (i, sig) in sig_e.iter_mut().enumerate() {
+        let _sig_i = signature.next().ok_or_else(|| {
+            OpcodeResolutionError::BlackBoxFunctionFailed(
+                BlackBoxFunc::SchnorrVerify,
+                format!("sig_e should be 32 bytes long, found only {i} bytes"),
+            )
+        })?;
+        let sig_i = witness_to_value(initial_witness, _sig_i.witness)?;
+        *sig = *sig_i.to_be_bytes().last().ok_or_else(|| {
+            OpcodeResolutionError::BlackBoxFunctionFailed(
+                BlackBoxFunc::SchnorrVerify,
+                "could not get last bytes".into(),
+            )
+        })?;
+    }
```

Comment:
@kevaundray My question on slack is basically around these lines. We restrict the public key to be a pair of elements of the underlying curve field so it stands to reason that the signature would be the same (and so can be represented as a pair of field elements) but we previously allowed a full u256 range.

Reply:
Closing as this is now a concern of #386